olaf/scripts/runner.py

```diff
@@ -0,0 +1,52 @@
+import os
+import spacy
+from abc import ABC, abstractmethod
+from olaf import Pipeline
+from olaf.repository.corpus_loader import TextCorpusLoader
+from olaf.repository.serialiser.kr_serialisers import KRJSONSerialiser
+
+
+class Runner(ABC):
+    def __init__(self, model_name: str, corpus_path: str):
+        """Initialise a pipeline Runner.
+
+        Attributes
+        ----------
+        pipeline: Pipeline
+            The pipeline to execute.
+        """
+        spacy_model = spacy.load(model_name)
+        corpus_loader = TextCorpusLoader(corpus_path=corpus_path)
+        self.pipeline = Pipeline(spacy_model=spacy_model, corpus_loader=corpus_loader)
+
+    @abstractmethod
+    def add_pipeline_components(self) -> None:
+        """Add all neccesary components in the pipeline"""
+
+    def run(self, name: str) -> None:
+        """LLM pipeline execution."""
+
+        self.add_pipeline_components()
+        self.pipeline.run()
+
+        kr_serialiser = KRJSONSerialiser()
+        kr_serialisation_path = os.path.join(os.getcwd(), f"{name}_kr.json")
+        kr_serialiser.serialise(kr=self.pipeline.kr, file_path=kr_serialisation_path)
+
+        kr_rdf_graph_path = os.path.join(
+            os.getcwd(),
+            f"{name}_kr_rdf_graph.ttl",
```

Comment:
this will create the results in the scripts folder but it would be better to have it in the data folder

Reply:
ok why not